pkg/systemd/overview-cards/cryptoPolicies.jsx

```diff
@@ -0,0 +1,230 @@
+    const [currentCryptoPolicy, setCurrentCryptoPolicy] = useState("");
+    const [isOpen, setIsOpen] = useState(false);
+    const [permission, setPermission] = useState(null);
+
+    useEffect(() => {
+        // Avoid cockpit.spawn every re-render from parent
+        if (!showCryptoPolicy) {
+            cockpit.spawn(["update-crypto-policies", "--show"], { err: "message" }).then(output => {
+                setCurrentCryptoPolicy(output.trim());
+                setShowCryptoPolicy(true);
+            });
+            // When users run `update-crypto-policies` manually.
+            cockpit.file("/etc/crypto-policies/state/current").watch(content => {
+                if (content) {
+                    setCurrentCryptoPolicy(content.trim());
+                    setShowCryptoPolicy(true);
+                }
+            });
+            setPermission(cockpit.permission({ admin: true }));
+        }
+    }, [showCryptoPolicy]);
+
+    if (!showCryptoPolicy) {
+        return null;
+    }
+
+    return (
+        <tr>
+            <th scope="row">{_("Crypto policy")}</th>
+            <td>
+                <PrivilegedButton variant="link" buttonId="crypto-policy-button" tooltipId="tip-crypto-policy"
+                                excuse={ _("The user $0 is not permitted to change crypto policies") }
+                                permission={ permission } onClick={() => setIsOpen(true)}>
+                    {displayProfileText(currentCryptoPolicy)}
+                </PrivilegedButton>
+                {isOpen &&
+                <CryptoPolicyDialog close={() => setIsOpen(false)}
+                          currentCryptoPolicy={currentCryptoPolicy}
+                          setCurrentCryptoPolicy={setCurrentCryptoPolicy}
+                />
+                }
+            </td>
+        </tr>
+    );
+};
+
+const CryptoPolicyDialog = ({
+    close,
+    currentCryptoPolicy,
+    setCurrentCryptoPolicy,
+}) => {
+    const [error, setError] = useState();
+    const [selected, setSelected] = useState(currentCryptoPolicy);
+
+    const setPolicy = (reboot) => {
+        cockpit.spawn(["update-crypto-policies", "--set", selected], { superuser: "require", err: "message" })
+                .then(() => {
+                    setCurrentCryptoPolicy(selected);
+                    if (reboot) {
+                        cockpit.spawn(["shutdown", "--reboot", "now"], { superuser: "require", err: "message" })
+                                .catch(error => setError(error));
+                    } else {
+                        cockpit.file(stampfile, { superuser: "require" }).replace("\n")
+                                .then(() => {
+                                    close();
+                                });
+                    }
+                })
+                .catch(error => setError(error));
+    };
+
+    const policies = Object.keys(cryptopolicies).map(policy => {
+        return {
```

Comment:
Please replace `{ return ... }` code block with `({ ... })` to get a pure data mapping